builder/dockerfile/parser/parser.go

```diff
@@ -152,8 +153,14 @@ func Parse(rwc io.Reader) (*Node, error) {
 	root.StartLine = -1
 	scanner := bufio.NewScanner(rwc)
 
+	utf8bom := []byte{0xEF, 0xBB, 0xBF}
 	for scanner.Scan() {
-		scannedLine := strings.TrimLeftFunc(scanner.Text(), unicode.IsSpace)
+		scannedBytes := scanner.Bytes()
+		// We trim UTF8 BOM
+		if currentLine == 0 && bytes.HasPrefix(scannedBytes, utf8bom) {
```

Comment:
Thanks @thaJeztah The pull request has been updated to remove the `bytes.HasPrefix()`.